tensorflow/contrib/seq2seq/python/ops/decoder.py

```diff
@@ -112,6 +112,20 @@ def step(self, time, inputs, state, name=None):
     raise NotImplementedError
 
   def finalize(self, outputs, final_state, sequence_lengths):
+    """Called after any decoding iterations.
+
+    Args:
+      outputs: RNNCell outputs (possibly nested tuple of) tensor[s] for all time
+        steps.
+      final_state: RNNCell final state (possibly nested tuple of) tensor[s] for
+        last time step.
+      sequence_lengths: Scalar `int32` tensor. Maximum sequence length.
```

Comment:
I don't think this is the maximum.  It is a vector of lengths

Reply:
```python3
#!/usr/bin/env python
'''
Seq2Seq implementation, practice
Example of tf.contrib.dynamic_decode
'''

import numpy as np
import tensorflow as tf
import tensorflow.contrib.seq2seq as seq2seq

class Decoder(seq2seq.Decoder):

    def __init__(self, batch_size, initial_input, initial_state):
        self._batch_size = batch_size
        self._initial_input = initial_input
        self._initial_state = initial_state

    @property
    def batch_size(self):
        return self._batch_size
    
    @property
    def output_dtype(self):
        return tf.float32
    
    @property
    def output_size(self):
        return 3
    
    @property
    def tracks_own_finished(self):
        return True

    def finalize(self, outputs, final_state, sequence_lengths):
        final_outputs = outputs
        final_state = final_state
        print('finalize: sequence_lengths:', sequence_lengths)
        return final_outputs, final_state
    
    def initialize(self, name=None):
        finished = False
        initial_inputs = self._initial_input
        initial_state = self._initial_state
        return finished, initial_inputs, initial_state
    
    def step(self, time, inputs, state, name=None):
        outputs = state + inputs
        next_state = outputs
        next_inputs = outputs
        finished = tf.reduce_sum(outputs) >= 100
        return outputs, next_state, next_inputs, finished

def main():

    initial_input = tf.constant([[1,2,3], [2,3,4]], dtype=tf.float32)
    initial_state = tf.constant([[1,2,3], [2,3,4]], dtype=tf.float32)
    decoder = Decoder(2, initial_input, initial_state)
    outputs, state, seq_lengths = seq2seq.dynamic_decode(decoder)

    with tf.Session() as sess:
        sess.run(tf.global_variables_initializer())
        print('outputs:', sess.run(outputs))

if __name__ == '__main__':
    main()
```

I tried to make a sample code to know what is the `sequence_lengths`, the argument of `finalize`.

Below is result of code-run
```
finalize: sequence_lengths: Tensor("decoder/while/Exit_5:0", shape=(), dtype=int32)
outputs: [[[ 2.  4.  6.]
  [ 4.  8. 12.]
  [ 8. 16. 24.]]

 [[ 4.  6.  8.]
  [ 8. 12. 16.]
  [16. 24. 32.]]]
```
Above shows `sequence_lengths.shape` is ()

I think there may be other options to make finalize get `sequence_lengths` as ndarray of `int32`, but I can't find it.

If you don't mind, can you give me some hint, or help?

Reply:
I apologize you if you wasted your time to read above reply...

I found the way to get `sequence_lengths` as ndarray of `int32`.

I'm sorry